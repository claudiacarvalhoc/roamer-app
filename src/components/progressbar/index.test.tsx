import React from 'react';
import ProgressBar, { ProgressBarProps } from './index';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

const getShallowWrapper = (props) => {
    return shallow(<ProgressBar {...props} />);
};

describe('<ProgressBar />', () => {
    it('render correctly with small size', () => {
        const props: ProgressBarProps = {
            className: 'test',
            percentage: 100,
            size: 'small'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly with medium size', () => {
        const props: ProgressBarProps = {
            className: 'test',
            percentage: 100,
            size: 'medium'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly with percentage lower then 20', () => {
        const props: ProgressBarProps = {
            className: 'test',
            percentage: 10,
            size: 'small'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly with percentage lower then 40', () => {
        const props: ProgressBarProps = {
            className: 'test',
            percentage: 35,
            size: 'medium'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
    it('render correctly with percentage greater then 80', () => {
        const props: ProgressBarProps = {
            className: 'test',
            percentage: 90,
            size: 'medium'
        };
        const component = getShallowWrapper(props);
        expect(toJson(component)).toMatchSnapshot();
    });
});
