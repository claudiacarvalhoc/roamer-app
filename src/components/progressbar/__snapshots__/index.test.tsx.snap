// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> render correctly with medium size 1`] = `
<div
  className="test"
>
  <div
    style={
      Object {
        "border": "2px solid #CFCFCF",
        "height": 0,
        "width": "205px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "2px solid #5489DC",
          "margin-left": "-2px",
          "margin-top": "-2px",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> render correctly with percentage greater then 80 1`] = `
<div
  className="test"
>
  <div
    style={
      Object {
        "border": "2px solid #CFCFCF",
        "height": 0,
        "width": "205px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "2px solid #5489DC",
          "margin-left": "-2px",
          "margin-top": "-2px",
          "width": "90%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> render correctly with percentage lower then 20 1`] = `
<div
  className="test"
>
  <div
    style={
      Object {
        "border": "2px solid #CFCFCF",
        "height": 0,
        "width": "160px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "2px solid #DB1F35",
          "margin-left": "-2px",
          "margin-top": "-2px",
          "width": "10%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> render correctly with percentage lower then 40 1`] = `
<div
  className="test"
>
  <div
    style={
      Object {
        "border": "2px solid #CFCFCF",
        "height": 0,
        "width": "205px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "2px solid #FFC400",
          "margin-left": "-2px",
          "margin-top": "-2px",
          "width": "35%",
        }
      }
    />
  </div>
</div>
`;

exports[`<ProgressBar /> render correctly with small size 1`] = `
<div
  className="test"
>
  <div
    style={
      Object {
        "border": "2px solid #CFCFCF",
        "height": 0,
        "width": "160px",
      }
    }
  >
    <div
      style={
        Object {
          "border": "2px solid #5489DC",
          "margin-left": "-2px",
          "margin-top": "-2px",
          "width": "100%",
        }
      }
    />
  </div>
</div>
`;
